plugins/ksp/src/org/jetbrains/kotlin/ksp/processor/JavaModifierProcessor.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.ksp.processor
+
+import org.jetbrains.kotlin.ksp.processing.Resolver
+import org.jetbrains.kotlin.ksp.symbol.*
+import org.jetbrains.kotlin.ksp.visitor.KSTopDownVisitor
+
+class JavaModifierProcessor : AbstractTestProcessor() {
+    val results = mutableListOf<String>()
+
+    override fun toResult(): List<String> {
+        return results
+    }
+
+    override fun process(resolver: Resolver) {
+        val symbol = resolver.getSymbolsWithAnnotation("Test").single() as KSClassDeclaration
+        symbol.superTypes.single().resolve()!!.declaration.accept(ModifierVisitor(), Unit)
+    }
+
+    inner class ModifierVisitor : KSTopDownVisitor<Unit, Unit>() {
+        override fun defaultHandler(node: KSNode, data: Unit) {
+        }
+
+        override fun visitClassDeclaration(classDeclaration: KSClassDeclaration, data: Unit) {
+            this.visitModifierListOwner(classDeclaration, data)
+            classDeclaration.declarations.map { it.accept(this, data) }
+        }
+
+        override fun visitPropertyDeclaration(property: KSPropertyDeclaration, data: Unit) {
+            this.visitModifierListOwner(property, data)
+        }
+
+        override fun visitFunctionDeclaration(function: KSFunctionDeclaration, data: Unit) {
+            this.visitModifierListOwner(function, data)
+        }
```

Comment:
These 3 overrides (`visitClassDeclaration`, `visitPropertyDeclaration`, `visitFunctionDeclaration`) seem unnecessary.

Reply:
one example is getQualifier is not implemented for Java symbols yet, so overriding them can avoid running into not implemented operations in TopDown visitor.